impl/src/main/java/org/pentaho/di/trans/dataservice/www/ListDataServicesServlet.java

```diff
@@ -71,19 +71,35 @@ public void handleRequest( CarteRequest request ) throws IOException {
 
     request.respond( 200 )
       .with( "text/xml; charset=utf-8", new WriterResponse() {
+        private void writeServiceXml( PrintWriter writer, IThinServiceInformation thinServiceInformation ) throws IOException {
+          writer.println(XMLHandler.openTag(XML_TAG_SERVICE));
+
+          writer.println(XMLHandler.addTagValue("name", thinServiceInformation.getName()));
+          writer.println(XMLHandler.addTagValue("streaming", thinServiceInformation.isStreaming()));
+          writer.println(thinServiceInformation.getServiceFields().getMetaXML());
+
+          writer.println(XMLHandler.closeTag(XML_TAG_SERVICE));
+        }
+
         @Override
         public void write( PrintWriter writer ) throws IOException {
           writer.println( XMLHandler.getXMLHeader() );
           writer.println( XMLHandler.openTag( XML_TAG_SERVICES ) );
 
           for ( IThinServiceInformation thinServiceInformation : serviceInformation ) {
-            writer.println( XMLHandler.openTag( XML_TAG_SERVICE ) );
 
-            writer.println( XMLHandler.addTagValue( "name", thinServiceInformation.getName() ) );
-            writer.println( XMLHandler.addTagValue( "streaming", thinServiceInformation.isStreaming() ) );
-            writer.println( thinServiceInformation.getServiceFields().getMetaXML() );
+            String streamingParam = request.getParameter( "streaming" );
+
+            if (streamingParam == null) {
+              writeServiceXml( writer, thinServiceInformation );
+            } else {
+              boolean streaming = Boolean.parseBoolean( request.getParameter("streaming") );
 
-            writer.println( XMLHandler.closeTag( XML_TAG_SERVICE ) );
+              if ((streaming && thinServiceInformation.isStreaming()) ||
```

Comment:
streaming == thinServiceInformation.isStreaming() ???? but don't bother to push again if no other change is needed... it's correct any way